### DISCUSSION TITLE
File Upload not works

### DISCUSSION THREAD

**Discussion by gctom** [livewire/livewire#8303] [1. Help]
Hi, experts.

As Livewire 3 documentation mentions file upload I followed all as it is.

I created route:
Route::any('/upload', Upload::class);

class code:
namespace App\Livewire;

use Livewire\Component;
use Livewire\WithFileUploads;
use Livewire\Attributes\Validate;
class Upload extends Component
{

    use WithFileUploads;
 
    #[Validate('image|max:1024')] // 1MB Max
    public $photo;
 
    public function save()
    {
        $this->photo->store(path: 'photos');
    }
    public function render()
    {
        return view('livewire.upload');
    }

}

and blade:
<form wire:submit="save">

        <input type="file" wire:model.defer="photo">
     
        @error('photo') <span class="error">{{ $message }}</span> @enderror
        <div wire:loading wire:target="photo">Uploading...</div>
        <button type="submit">Save photo</button>
    </form>
    
    #problem
    
    while I choose the image for upload then error message display = "The photo failed to upload." automatically without click on submit button.
    
    In console shows error
     "POST http://livewire.test/livewire/upload-file?expires=1713096082&signature=a37ba1e5f38b5310b42c43ae2f30e838afd3ab7c83f0cab308a8dedd65657dcc 500 (Internal Server Error)"
     
     If I click this console error then shows on network header:
     Request URL:
http://livewire.test/livewire/upload-file?expires=1713096082&signature=a37ba1e5f38b5310b42c43ae2f30e838afd3ab7c83f0cab308a8dedd65657dcc
Request Method:
POST
Status Code:
500 Internal Server Error
Remote Address:
127.0.0.1:80
Referrer Policy:
strict-origin-when-cross-origin

in preview tab:
{message: "Path cannot be empty", exception: "ValueError",…}
exception
: 
"ValueError"
file
: 
"C:\\laragon\\www\\livewire\\vendor\\laravel\\framework\\src\\Illuminate\\Filesystem\\FilesystemAdapter.php"
line
: 
414
message
: 
"Path cannot be empty"

In Response Tab:
  "message": "Path cannot be empty",
  
  
  
  what is the exactly problem and how can I solve this?
  
  Laravel 11, livewire 3

**Comment by KgsMIsmail** [comment]
How many upload components are there in your project?
In my case, there are two upload components on the page, and I didn't realize that the property names for both upload components identical. Then I changed the property name for each component and the error disappeared, everything went smoothly.

**Comment by samlev** [comment]
Try removing `.defer` from your `wire:model`. It's the default behaviour in livewire v3, but more than that livewire uploads work in a couple of steps - when you select the file, it will automatically upload it, then once it has finished uploading you can save it.

I suspect that what's happening is that your image hasn't finished uploading before you press "Save Photo", so the attempt to move the file fails.

Try updating your submit button:
```html
<button type="submit" @disabled(empty($photo))>
  Save Photo
</button>
```

**Comment by Fardan-Nurhidayat** [comment]
Hello , in here i have a same problem , but i have one components for upload. try open your php.ini and then set on in allow_url_include like this allow_url_include = On, in  my case that works

**Comment by VenuSathish321** [comment]
Hello ,I am also same proble while uplaod file in laravel 12 livewire 3 I changed allow_url_include = On @Fardan-Nurhidayat but still not working kindly helpme
![image](https://github.com/user-attachments/assets/ef8af324-11c9-419a-997a-569b9c1f8bf9)

**Reply by Fardan-Nurhidayat** [reply]
It could be help you , go to php.ini then uncomment upload_tmp_dir = c:\windows\temp.

You can go to https://stackoverflow.com/questions/70093647/path-cannot-be-empty-laravel

For more detail

**Reply by VenuSathish321** [reply]
@Fardan-Nurhidayat Now Working fine for me thank you for your reply

**Reply by drdan18** [reply]
Thank you.  This just solved a 3 hour headache.
